paderbox/io/download.py

```diff
@@ -0,0 +1,184 @@
+
+import os
+import socket
+import sys
+import tarfile
+import zipfile
+from os import path
+from urllib.request import urlretrieve
+
+from tqdm import tqdm
+
+
+def download_file(remote_file, local_file):
+    """
+    Download single file to local_dir
+
+    Args:
+        remote_file:
+        local_file:
+
+    Returns:
+
+    """
+    local_dir = path.dirname(local_file)
+    if not path.isfile(local_file):
+        def progress_hook(t):
+            """
+            Wraps tqdm instance. Don't forget to close() or __exit__()
+            the tqdm instance once you're done with it (easiest using
+            `with` syntax).
+            """
+
+            last_b = [0]
```

Comment:
How about using `nonlocal` in `inner`? This would be a bit more explicite.

```python
last_b = 0
def inner(...):
  nonlocal last_b
  ...
  last_b = ...
```

```